source/slang/slang-ast-expr.h

```diff
@@ -176,6 +176,27 @@ class StaticMemberExpr: public DeclRefExpr
     RefPtr<Expr> baseExpression;
 };
 
+struct MatrixCoord
+{
+    bool operator==(const MatrixCoord rhs) const {
```

Comment:
For a short single line expression like this it's ok to just do as 
```
bool operator==(const MatrixCoord& rhs) const { return row == rhs.row && col == rhs.col; };
```

You might want to make the rhs a reference.